File: Python/Medium/1d-programming/largestDivisableSubset.py
def solution(nums):
    subsets = {-1: set()}

    nums.sort()
    for num in nums:
        divisors = []
        for key in subsets.keys():
            if num % key == 0:
                divisors.append(key)

        possibilities = [subsets[possibility] for possibility in divisors]

        longest_subset = max(possibilities, key = len)
        subsets[num] = longest_subset | {num}

    return max(subsets.values(), key = len)

File: JavaScript/Medium/1d_programming/largestDivisableSubset.js
var solution = (nums) => {
    let subsets = new Map()
    subsets.set(-1, new Set())

    nums.sort((a,b) => a-b)

    for (const num of nums){
        let divisors = []
        for (const [key, value] of subsets){
            if (num % key === 0){
                divisors.push(value)
            }
        }

        let maxSize = 0
        let maxSubset = new Set()
        for (let subset of divisors){
            if (subset.size > maxSize) {
                maxSize = subset.size
                maxSubset = new Set(subset)
            }
        }

        maxSubset.add(num)
        subsets.set(num, maxSubset)
    }

    let res = []
    let length = 0
    for (let arr of subsets.values()){
        if (arr.size > length) {
            length = arr.size
            res = Array.from(arr)
        }
    }

    return res
}